analytics/upress_reporting/counter_5_controller.py

```diff
@@ -1,74 +1,198 @@
 import argparse
-from ast import parse
 import os
 import pandas
-import re
 
 from datetime import datetime
-from models.pollers.download_data_poller import DownloadDataPoller
-from models.pollers.view_data_poller import ViewDataPoller
+from helpers import aggregate_logs
+from helpers.format_data import format_to_interaction_event
+from managers.db import DBManager
+from models.data.interaction_event import InteractionType
+from models.pollers.poller import Poller
+from model.postgres.edition import Edition
+from model.postgres.item import Item
+from model.postgres.record import Record
 from models.reports.country_level import CountryLevelReport
 from models.reports.downloads import DownloadsReport
 from models.reports.total_usage import TotalUsageReport
 from models.reports.views import ViewsReport
+from sqlalchemy import CTE, func, literal
+from typing import List
+
+VIEW_FILE_ID_REGEX = r"REST.GET.OBJECT manifests/(.*?json)\s"
+DOWNLOAD_FILE_ID_REGEX = r"REST.GET.OBJECT (.+pdf\s)"
 
 
 class Counter5Controller:
     def __init__(self, args):
         self.publishers = os.environ.get("PUBLISHERS").split(",")
+
         parsed_args = self._parse_args(args)
         if parsed_args is not None:
             self.reporting_period = self._parse_reporting_period(parsed_args)
         else:
-            self.reporting_period = (f"{datetime.now().year}-01-01 to {datetime.now().year}-01-31")
+            self.reporting_period = f"{
+                datetime.now().year}-01-01 to {datetime.now().year}-01-31"
+
+        self.view_bucket = os.environ.get("VIEW_BUCKET", None)
+        self.download_bucket = os.environ.get("DOWNLOAD_BUCKET", None)
+        self.referrer_url = os.environ.get("REFERRER_URL", None)
+
+        self.setup_db_manager()
+
+    def setup_db_manager(self):
+        self.db_manager = DBManager(
+            user=os.environ.get("POSTGRES_USER", None),
+            pswd=os.environ.get("POSTGRES_PSWD", None),
+            host=os.environ.get("POSTGRES_HOST", None),
+            port=os.environ.get("POSTGRES_PORT", None),
+            db=os.environ.get("POSTGRES_NAME", None),
+        )
+        self.db_manager.generateEngine()
+        self.db_manager.createSession()
 
     def create_reports(self):
-        print("Generating Counter 5 reports...")
+        print("Generating Counter 5 reports...", datetime.now())
+        # TODO: aggregate log files -> * one for views and one for downloads (this is agnostic of publisher)
```

Comment:
Are these TODOs done? If so, let's remove the comments - see below as well.

Reply:
Yep, they're done! Just left them in to illustrate how I addressed our original plan for improving performance. They are removed now